addons/popochiu/engine/objects/character/popochiu_character.gd

```diff
@@ -811,14 +810,21 @@ func _get_valid_oriented_animation(animation_label):
 	return null
 
 
-func _walk_to_node(node: Node2D, offset: Vector2) -> void:
+func _walk_to_node(node: Node2D, offset: Vector2, is_blocking := false) -> void:
 	if not is_instance_valid(node):
 		await get_tree().process_frame
 		return
-
-	await walk(
+	
+	var target_pos := (
 		node.to_global(node.walk_to_point if node is PopochiuClickable else Vector2.ZERO) + offset
 	)
+	var target_move_pos: Vector2 = R.current.get_move_target_position(position, target_pos)
+	
+	await walk(target_pos)
+	
+	if not is_blocking and position != target_move_pos:
```

Comment:
I'm probably misreading this, but shouldn't it be:

"If IS BLOCKING and you are not yet at destination, then await the movement is over" ?

Reply:
Nope.

The purpose of this `if` statement is to verify that the final position of the movement is as expected only when the action is NON-BLOCKING. Otherwise, there is no need to perform the validation because it is assumed that the character will reach its destination since the GUI will be blocked, preventing clicks from being registered.

Reply:
Ah OK, got it.
Thanks